testdata/Makefile

```diff
@@ -29,3 +41,42 @@ loader-%-eb.elf: loader.c
 
 %-eb.elf : %.c
 	$(CLANG) $(CFLAGS) -mbig-endian -c $< -o $@
+
+help: ## Display this help.
+	@grep -E '^[a-zA-Z_-]+:.*## .*$$' $(MAKEFILE_LIST) | \
+		awk 'BEGIN {FS = ":.*## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+
+else
+
+# If the first make goal contains `docker`, only declare Docker-related targets.
+
+.PHONY: docker docker-shell
+
+# Get the absolute path and uid/gid of the repository on disk.
+GITDIR := $(shell git rev-parse --show-toplevel)
+UIDGID := $(shell stat -c '%u:%g' ${GITDIR})
+
+# Pop off the 'docker' goal so the remaining goals can be passed to
+# the Dockerized make.
+PARAMS := $(filter-out docker,$(MAKECMDGOALS))
+
+DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+REPO := $(shell cat ${DIR}/docker/REPO)
+VERSION := $(shell cat ${DIR}/docker/VERSION)
```

Comment:
Why not `latest`?

Reply:
Having only `latest` would defeat the point of this change entirely, since it would make ELFs in historical commits non-reproducible. IMO having versioned containers is a hard requirement here. A possible next step could even be running `make` in testdata during CI to make sure the ELFs correspond to the C code.